test/distributed/org/apache/cassandra/distributed/test/ring/BootstrapTest.java

```diff
@@ -154,9 +163,58 @@ public void bootstrapJMXStatus() throws Throwable
                 assertEquals("COMPLETED", StorageService.instance.getBootstrapState());
                 assertFalse(StorageService.instance.isBootstrapFailed());
             });
+
+            assertEquals(Long.valueOf(0L), getMetricGaugeValue(joiningInstance, "BootstrapFilesTotal", Long.class));
+            assertEquals(Long.valueOf(0L), getMetricGaugeValue(joiningInstance, "BootstrapFilesReceived", Long.class));
+            assertEquals("Bootstrap streaming success", getMetricGaugeValue(joiningInstance, "BootstrapLastSeenStatus", String.class));
+            assertEquals("", getMetricGaugeValue(joiningInstance, "BootstrapLastSeenError", String.class));
         }
     }
 
+    public static <T> T getMetricGaugeValue(IInvokableInstance instance, String metricName, Class<T> gaugeReturnType)
+    {
+        return gaugeReturnType.cast(getMetricAttribute(instance, metricName, "Value"));
+    }
+
+    public static long getMetricMeterRate(IInvokableInstance instance, String metricName)
+    {
+        Object raw = getMetricAttribute(instance, metricName, "Count");
+        return raw == null ? 0 : (Long) raw;
+    }
+
+    public static Object getMetricAttribute(IInvokableInstance instance, String metricName, String attributeName)
+    {
+        if (instance.isShutdown())
+            throw new IllegalStateException("Instance is shutdown");
+
+        try (JMXConnector jmxc = JMXUtil.getJmxConnector(instance.config()))
+        {
+            MBeanServerConnection mbsc = jmxc.getMBeanServerConnection();
+            ObjectName metric = mbsc.queryNames(null, null)
+                    .stream()
```

Comment:
would you mind to align all of this under .queryNames, please?

Reply:
Fixed.